filter.c

```diff
@@ -26,31 +52,34 @@ int filter(struct xdp_md *ctx)
 	struct iphdr *ip = data + sizeof(struct ethhdr);
 
 	if (ip + 1 > (struct iphdr *)data_end)
-		goto pass;
+		return XDP_DROP;
```

Comment:
it's weird for me to define `DOME_XDP_DENY` and return `XDP_DROP` here.. but I would `XDP_DROP` directly in the other places instead of changing this line..

Reply:
I didn't understand what do you mean by "in other places", which ones?

Notice that I removed `*_DENY` everywhere, so it's at least not defined anymore; I'm not sure whether this settles the matter for you or not.

Just to make it clear: the reason why we `drop` now instead of accepting, is because accepting an invalid ethernet packet now would incur in a fault, as we need to swap the ethernet headers. I didn't see a reason to accept invalid ethernet.

Also, I just noticed it may be an oversimplification to check if it's a valid IP when we actually want to check the ethernet here.